File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.swift
func first_and_last (element: Int, list: inout Array<Int>) -> Array<Int>{
	var find_first: Bool = true;
	var out: Array<Int> = [-1, -1];
	let len_list: Int = list.count;
	list.sort();
	
	for i in 0..<len_list{
		if (element == list[i]){
			if (find_first){
				out[0] = i;
				out[1] = i;
				find_first = false;
			} else {
				out[1] = i;
			}
		}
	}
	
	return (out);
}

var nums: Array<Int> = [5,7,7,8,8,10];
let target: Int = 8;

let positions: Array<Int> = first_and_last(element: target, list: &nums);

print("\n\nThis is the handy work of Swift\n\n\n");
print(nums);

print("\n\n\n\n");
print(positions);
print("\n\n\n\n");

File: 34. Find First and Last Position of Element in Sorted Array/find_first_and_last_position_of_element_in_sorted_array.js
function firstAndLast (element, list){
	let findFirst = true;
	let out = [-1,-1];
	list.sort(function (a, b) {return a - b});
	//console.log(list);
	
	for (let i = 0; i < list.length; i++){
		let e = list[i];
		if (e == element){
			if (findFirst){
				out[0] = i;
				out[1] = i;
				findFirst = false;
			} else {
				out[1] = i;
			}
		}
	}
	
	return (out);
}

let nums = [5,7,7,8,8,10,2];
let target = 8;

console.log(firstAndLast (target, nums));
